pr

mctapok opened the pull request.

mctapok committed:
pr

mctapok committed:
рефаткор кода
считывание команд вынесено в отдельный метод
обработка команд вынесена в отдельные методы
доработана логика работы таймеров
добавлен логгер
добавлен пропертис для бд
исправлена работа постоянного бана

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/ClientHandler.java:
```diff
@@ -0,0 +1,239 @@
+package ru.gavrilov.course_project.chat.server;
```
Здесь и далее в других классах: имя пакета не должно содержать нижнее подчеркивание _

```
package ru.gavrilov.courseproject.chat.server;
```

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/ClientHandler.java:
```diff
@@ -0,0 +1,239 @@
+        }).start();
+    }
+
+    private void listenUserChatMessages() throws IOException {
```
Метод лучше отрефакторить и вынести из него каждую конструкцию типа `if (message.equals("/exit"))` в отдельный приватный метод класса. Рекомендуется сделать общую длину listenUserChatMessages не более 15 строк

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/ClientHandler.java:
```diff
@@ -0,0 +1,239 @@
+            }
+            userActivityMonitor.resetTimer(socket);
+            if (message.contains("disconnected")) {
+                System.out.println("client " + socket.getPort() + " was disconnected ");
```
Здесь и далее все System.out. заменить на Logger

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/ClientHandler.java:
```diff
@@ -0,0 +1,239 @@
+    }
+
+    private void listenUserChatMessages() throws IOException {
+        while (true) {
```
Многовато break и continue в этом цикле. Надо попробовать сократить

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/ClientHandler.java:
```diff
@@ -0,0 +1,239 @@
+                            continue;
+                        }
+                        sendMessage("ERROR: time is not correct");
+                        continue;
```
Мне кажется, что `continue` здесь не нужен. Из-за того, что метод очень большой - он сложно читается.

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/DataBase.java:
```diff
@@ -0,0 +1,128 @@
+
+    public boolean dbConnection() {
+        try {
+            connection = DriverManager.getConnection(DATABASE_URL, "postgres", "postgres");
```
Хорошо бы пароль здесь не показывать в явную, а получать из файла с настройками

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/DataBase.java:
```diff
@@ -0,0 +1,128 @@
+    public User dbAuthorization(String login, String password) {
+        String selectQuery = "select login, username, password, isAdmin from users where login=? and password=?";
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(selectQuery);
```
Используйте try-with-resources для PreparedStatement

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/DataBase.java:
```diff
@@ -0,0 +1,128 @@
+            PreparedStatement preparedStatement = connection.prepareStatement(selectQuery);
+            preparedStatement.setString(1, login);
+            preparedStatement.setString(2, password);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
```
Здесь и далее: вложенность блоков try/catch не лучшим образом влияет на читаемость кода. Лучше внутренний блок try вынести в приватный метод класса

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/DataBase.java:
```diff
@@ -0,0 +1,128 @@
+import java.sql.*;
+
+public class DataBase {
+    private final String DATABASE_URL = "jdbc:postgresql://localhost:5432/course_project";
```
```
    private static final String DATABASE_URL = "jdbc:postgresql://localhost:5432/course_project";
```

и само значение тоже неплохо было бы не хардкодить, а читать из файла с настройками

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/Server.java:
```diff
@@ -0,0 +1,212 @@
+            clientHandler.sendMessage("ERROR: user not found in ban list");
+            return;
+        }
+        clientHandler.sendMessage("SERVER: admin command, permission denied");
```
Литерал "SERVER: admin command, permission denied" встречается 4 раза - его необходимо переопределить как статическую константу класса и заменить в 4-х местах

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/Server.java:
```diff
@@ -0,0 +1,212 @@
+
+        try {
+            serverSocket = new ServerSocket(port);
+            System.out.printf("Сервер запущен на порту %d. Ожидание подключения клиентов\n", port);
```
1) Здесь и далее: заменить System.out. на Logger

2) `System.out.printf("Сервер запущен на порту %d. Ожидание подключения клиентов%n", port)`

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/User.java:
```diff
@@ -0,0 +1,32 @@
+
+public class User {
+    private String login;
+    private String password;
```
Не используем поле?

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/UserActivityMonitor.java:
```diff
@@ -0,0 +1,78 @@
+
+public class UserActivityMonitor {
+    private Timer timer;
+    private Socket socket;
```
Не используем поле?

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/UserActivityMonitor.java:
```diff
@@ -0,0 +1,78 @@
+            @Override
+            public void run() {
+                try {
+                    System.out.println(socket + " disconnected");
```
Здесь и далее: заменить на Logger

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/UserActivityMonitor.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    public void resetTimer(Socket socket) {
+        Timer timer = clientTimers.get(socket);
```
Мы объявляем переменную и одновременно у нас есть такое же поле в классе. Рекомендуется делать, чтобы имена отличались

sproshchaev commented on server/src/main/java/ru.gavrilov.course_project.chat.server/UserActivityMonitor.java:
```diff
@@ -0,0 +1,78 @@
+                        e.printStackTrace();
+                    }
+                }
+            }, 20 * 60 * 1000);
```
Лучше объявить выше `long delay = 20 * 60 * 1000;`
И подставить сюда `}, delay);` вместо `}, 20 * 60 * 1000);`

mctapok committed:
рефаткор кода
считывание команд вынесено в отдельный метод
обработка команд вынесена в отдельные методы
доработана логика работы таймеров
добавлен логгер
добавлен пропертис для бд
исправлена работа постоянного бана

mctapok commented:
сделан рефактор
